### PR TITLE
Merge datafield_stages branch

### PR DESCRIPTION
**Pull request by martwo**
This branch provides the definition of data fields that need to be loaded from the data files (exp and/or mc) and at which stages these data fields are required. See the documentation and/or CHANGELOG file for more info.

### CODE REVIEW

File: skyllh/core/datafields.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+"""This module provides functionality for defining which data fields of a data
+file is required at what stage.
+"""
+
+
+class DataFieldStages(
+    object,
+):
+
```
Comment (tomaskontrimas) [inline skyllh/core/datafields.py:11]: I would add a docstring, explaining that the following constants are constructed as a binary representation

File: skyllh/core/datafields.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    ):
+        """Returns the list of data field names that match at least one of the
+        given stages, i.e. the joint set of data fields given the stages.
+
+        Parameters
+        ----------
+        datafields : dict
+            The dictionary of datafield names as keys and stages as values.
+        stages : sequence of int
+            The stages for which data field names should
```
Comment (tomaskontrimas) [inline skyllh/core/datafields.py:54]: missing ending of the sentence

**Review by tomaskontrimas — CHANGES_REQUESTED** [review]

File: skyllh/core/datafields.py (hunk 1 above)
Comment (martwo) [reply skyllh/core/datafields.py:14]: Done.

File: skyllh/core/datafields.py (hunk 2 above)
Comment (martwo) [reply skyllh/core/datafields.py]: Done.

**Review by tomaskontrimas — APPROVED** [review]

**Pull request merged**